### DISCUSSION TITLE
Changing select() option's return type in useInfiniteQuery

### DISCUSSION THREAD

**Discussion by jeytii** [TanStack/query#2301] [Q&A]
Is there any way to tell Typescript that I want to return data with the following format...

```ts
{
    users: [...]
}
```

Instead of this format?
```ts
{
    pages: [
        { users: [...], nextCursor: 1 },
        { users: [...], nextCursor: 2 },
        // ...
    ],
    pageParams: [...]
}
```

The following doesn't work as TS says the return type doesn't resemble with `InfiniteData`:
```ts
const { ... } = useInfiniteQuery<{ users: UserType[]; nextCursor?: number }>(
    'users',
    getUsers,
    {
        select(data): { users: UserType[] } {
            // ...

            return {
                users: [...]
            }
        }
    },
);
```

**Comment by TkDodo** [comment]
I think you need to return `InfiniteData` because otherwise, you'll break `getNextPage`. See also: https://github.com/tannerlinsley/react-query/issues/1537

**Reply by jeytii** [reply]
You mean like this?
```ts
select(data): InfiniteData<{ users: UserType[] }> {
    // ...

    return {
        users: [...]
    }
}
```

Already tried that, but TS would still tell me to return data that has the same format as the default `InfiniteData` type (with `pages` and `pageParams` properties).

**Reply by TkDodo** [reply]
No, i mean like in the issue:

```
useInfiniteQuery(['posts', variables], fetchPosts, {
  getNextPageParam: lastPage => lastPage.nextPage,
  select: data => ({ ...data, pages: data.pages.flatMap(page => page.posts) })
})
```

**Reply by jeytii** [reply]
That's not the format I want to return.

**Reply by TkDodo** [reply]
I think it’s what useInfiniteQuery _needs_ in order to work.
